pkg/apis/core/validation/validation_podscheduling_test.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+func testPodScheduling(name, namespace string, spec core.PodSchedulingSpec) *core.PodScheduling {
+	return &core.PodScheduling{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: spec,
+	}
+}
+
+func TestValidatePodScheduling(t *testing.T) {
+	goodName := "foo"
+	goodNS := "ns"
+	goodPodSchedulingSpec := core.PodSchedulingSpec{}
+	now := metav1.Now()
+	ten := int64(10)
+
+	scenarios := map[string]struct {
+		isExpectedFailure bool
```

Comment:
But which fields should I exclude? The error details might be a candidate because the explanation for bad labels and names are very long. But if do that unconditionally, then I also loose coverage for cases where I may want to check that the details are right.

I switched to checking all fields. Better thorough than sorry... I used testify/assert, it does diffs and is just a single line.

Looking at the details made me reconsider the decision to report the full `ResourceClaimSchedulingStatus` as duplicated value: that might be too large, the name is enough.

Reply:
In general, I at least exclude the value

Reply:
Not excluding anything was a useful exercise to ensure that also the values that get reported are useful (for example, not doing a diff in ValidateImmutableField). I think I'll keep it like this.